File: Leetcode/maxProfitI.swift
func maxProfit(prices: [Int]) -> Int {
    if prices.isEmpty {
        return 0
    }
    var maxValue: Int = 0
    var minNum: Int = prices[0]
    for var i = 1; i < prices.count; i++ {
        maxValue = max(maxValue, prices[i] - minNum)
        minNum = min(minNum, prices[i])
    }
    return maxValue
}

func maxProfit2(prices: [Int]) -> Int {
    if prices.isEmpty {
        return 0
    }
    var maxValue: Int = 0;
    var maxPrice: Int = prices[prices.count - 1]
    for var i = prices.count - 2; i >= 0 ; i-- {
        maxValue = max(maxValue, maxPrice - prices[i])
        maxPrice = max(maxPrice, prices[i])
    }
    return maxValue
}

File: Leetcode/maxProfitI.java
public int maxProfitI(int[] prices) {
        int max = 0;
        if (prices == null || prices.length < 2) {
            return 0;
        }
        int minPrice = prices[0];
        for (int i = 1; i < prices.length; i++) {
            max = Math.max(max, prices[i] - minPrice);
            minPrice = Math.min(minPrice, prices[i]);
        }
        return max;
    }
    
    
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length < 2) {
            return 0;
        }
        int max = 0;
        int maxPrice = prices[prices.length - 1];
        for (int i = prices.length - 2; i>=0; i--) {
            max = Math.max(max, maxPrice - prices[i]);
            maxPrice = Math.max(maxPrice, prices[i]);
        }
        return max;
    }
